### PR TITLE
[tesla] Fix discovery of Tesla vehicles from account

### PR DESCRIPTION
**Pull request by kaikreuzer**
So far, vehicles that are asleep (which is the normal situation) are not discovered and thus not put into the Inbox, what is fairly unexpected for the user.
This change makes sure that every vehicle of the account is added to the Inbox. It defers the identification of the model to a later stage as this is only required for creating the Thing with all its channels.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/discovery/TeslaVehicleDiscoveryService.java [hunk 1]
```diff
@@ -81,31 +81,15 @@ public void deactivate() {
 
     @Override
     public void vehicleFound(Vehicle vehicle, VehicleConfig vehicleConfig) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/discovery/TeslaVehicleDiscoveryService.java:83]: Please add `@Nullable` to the parameter `vehicleConfig`.

File: bundles/org.openhab.binding.tesla/src/main/resources/OH-INF/i18n/tesla.properties [hunk 2]
```diff
@@ -206,7 +206,7 @@ channel-type.tesla.minavailabletemp.label = Minimum Temperature
 channel-type.tesla.minavailabletemp.description = Indicates the minimal inside temperature of the vehicle
 channel-type.tesla.mobileenabled.label = Mobile Enabled
 channel-type.tesla.mobileenabled.description = Indicates whether the vehicle can be remotely controlled
-channel-type.tesla.notenoughpower.label = Not Enought Power
+channel-type.tesla.notenoughpower.label = Not Enough Power
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tesla/src/main/resources/OH-INF/i18n/tesla.properties:209]: Please also update the typo in the XML file.

And as you added a new thing type, you need to add all the new information for this thing type in this file (the easiest way is to run the i18n tool).

**Review by lolodomo — CHANGES_REQUESTED** [review]
I was not aware of the existence of the ThingTypeMigrationService service.

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/discovery/TeslaVehicleDiscoveryService.java (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/discovery/TeslaVehicleDiscoveryService.java:83]: The whole class/binding does not yet have null annotations on it, I wouldn't want to do it as part of a bug fix.

File: bundles/org.openhab.binding.tesla/src/main/resources/OH-INF/i18n/tesla.properties (hunk 2 above)
Comment (kaikreuzer) [reply bundles/org.openhab.binding.tesla/src/main/resources/OH-INF/i18n/tesla.properties:259]: I always forget this. But I do not seem to be the only one - regenerating the files shows a hell of a lot new entries that are not all related to my changes. 😉

**Comment by kaikreuzer** [comment]
@lolodomo Thanks for the quick review - I have addressed your comments.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**